Unable to build Embedded stdlib when using Xcode 26.4 beta 1, TargetConditionals.h: unknown compiler

### Description

Unable to build Embedded stdlib when using Xcode 26.4 beta 1, TargetConditionals.h: unknown compiler

```
FAILED: [code=1] stdlib/public/Concurrency/CMakeFiles/embedded-concurrency-default-executor-armv7-none-none-eabi.dir/CooperativeGlobalExecutor.cpp.o 
...
In file included from .../swift/stdlib/public/Concurrency/CooperativeGlobalExecutor.cpp:41:
In file included from .../swift/include/swift/Runtime/Heap.h:25:
In file included from .../swift/include/swift/Runtime/Config.h:570:
.../Xcode26.4b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX26.4.sdk/usr/include/TargetConditionals.h:391:10: error: TargetConditionals.h: unknown compiler (see comment above)
  391 |         #error TargetConditionals.h: unknown compiler (see comment above)
      |          ^
1 error generated.
```

### Reproduction

```bash
export DEVELOPER_DIR=<path to Xcode 26.0 beta 1>
swift/utils/build-script --preset=buildbot_all_platforms,tools=RA,stdlib=RA
```


### Expected behavior

No error in compiling Embedded stdlib

### Environment

Xcode 26.4 beta 1

### Additional information

_No response_

--- a/stdlib/public/Concurrency/CMakeLists.txt
+++ b/stdlib/public/Concurrency/CMakeLists.txt
@@ -312,6 +312,15 @@ if(SWIFT_SHOULD_BUILD_EMBEDDED_STDLIB AND SWIFT_SHOULD_BUILD_EMBEDDED_CONCURRENC
         # Host is macOS with a macOS SDK. To be able to build the C++ Concurrency runtime for non-Darwin targets using the macOS SDK,
         # we need to pass some extra flags and search paths.
         set(extra_c_compile_flags -stdlib=libc++ -isystem${SWIFT_SDK_OSX_PATH}/usr/include/c++/v1 -isystem${SWIFT_SDK_OSX_PATH}/usr/include -D__APPLE__)
+        # Before Xcode 26.4, we were getting these macros from TargetConditionals.h
+        # now it's on us to define those to keep using the macOS SDK for non Darwin targets
+        if("${arch}" MATCHES "armv7")
+          list(APPEND extra_c_compile_flags -DTARGET_CPU_ARM=1)
+        elseif("${arch}" MATCHES "arm64")
+          list(APPEND extra_c_compile_flags -DTARGET_CPU_ARM64=1)
+        elseif("${arch}" MATCHES "x86_64")
+          list(APPEND extra_c_compile_flags -DTARGET_CPU_X86_64=1)
+        endif()
       endif()
     endif()
 

